pandas/core/indexing.py

```diff
@@ -2119,9 +2120,16 @@ def _setitem_with_indexer_missing(self, indexer, value):
             new_values = Series([value], dtype=new_dtype)._values
 
             if len(self.obj._values):
+                # GH#47677 handle enlarging with a scalar as a special case
```

Comment:
This is not the right place for this. You have to create new_values correctly instead of recreating it here

Reply:
I do, here is the value of `new_values`

```python
(Pdb++) new_values
['a']
Categories (1, object): ['a']
```
The point of avaoiding concat_compat is, that in there we explicitly check if the dtypes are the same, and because the categories aren't the same the concatenated series is cast to object. If we'd change that then this will also have an effect on `pd.concat([s, pd.Series(["a", "b"], dtype="category")])` or we create a very specific special case in `concat_compat` and check for length 1.

Reply:
No, this is not what I was talking about. But lets start over:

If we want to handle this in maybe_promote, we should return the dtype we get as input, not the string ``"categories"``. This loses all information we already got. No need to special case afterwards then.

If we decide against handling this in maybe_promote, we have to handle this earlier, e.g. in line 2120 at the latest, not here. tolist() is never a good idea, if you want to keep the dtype, this looses all precision information, e.g.

```
result = Series([1, 2, 3], dtype=pd.CategoricalDtype(categories=pd.Index([1, 2, 3], dtype="Int64")))
result.loc[3] = 2
```

this is converted into CategoricalDtype with ``int64`` not ``Int64``, same would go for ``int32``. It would get converted into ``int64`` too.

Reply:
Thanks a lot. Of course you're right, just returning the given dtype works fine. I updated the PR, handling this in `maybe_promote` feels right for me, checking for dtypes and returning them is done there a lot.